aiohttp/websocket.py

```diff
@@ -226,6 +226,14 @@ def do_handshake(method, headers, transport, protocols=()):
                 'Client protocols %r don’t overlap server-known ones %r',
                 protocols, req_protocols)
 
+    if origins:
+        origin = headers.get('ORIGIN', '').strip()
+        if origin not in origins:
```

Comment:
[It doesn't simply](http://tools.ietf.org/html/rfc6454#section-4) to just search origin header value in some list seeking for [comparison](http://tools.ietf.org/html/rfc6454#section-5):
0. Origin value should be split into scheme, host, port triple
1. Host need to be converted to lower case
2. If port is not specified, the default one should be picked from the specified scheme
3. And only now compare scheme/host/port triples